src/docmanager/index.ts

```diff
@@ -0,0 +1,947 @@
+   * as the global default.
+   * If a factory is already registered as a default for a given extension or
+   * as the global default, this factory will override the existing default.
+   */
+  registerWidgetFactory(factory: IWidgetFactory<Widget>, options: IWidgetFactoryOptions): IDisposable {
+    let name = options.displayName;
+    let exOpt = options as Private.IWidgetFactoryEx;
+    exOpt.factory = factory;
+    if (this._widgetFactories[name]) {
+      throw new Error(`Duplicate registered factory ${name}`);
+    }
+    this._widgetFactories[name] = exOpt;
+    if (options.defaultFor) {
+      for (let option of options.defaultFor) {
+        if (option === '.*') {
+          this._defaultWidgetFactory = name;
+        } else if (option in options.fileExtensions) {
+          this._defaultWidgetFactories[option] = name;
+        }
+      }
+    }
+    return new DisposableDelegate(() => {
+      delete this._widgetFactories[name];
+      if (this._defaultWidgetFactory === name) {
+        this._defaultWidgetFactory = '';
+      }
+      for (let opt in Object.keys(this._defaultWidgetFactories)) {
+        let n = this._defaultWidgetFactories[opt];
+        if (n === name) {
+          delete this._defaultWidgetFactories[opt];
+        }
+      }
+    });
+  }
+
+  /**
+   * Register a model factory.
+   *
+   * @param factory - The factory instance.
+   *
+   * @param options - The options used to register the factory.
+   *
+   * @returns A disposable used to unregister the factory.
+   *
+   * #### Notes
+   * If a factory with the given `name` is already registered, an error
+   * will be thrown.
+   */
+  registerModelFactory(factory: IModelFactory, options: IModelFactoryOptions): IDisposable {
+    let exOpt = options as Private.IModelFactoryEx;
+    let name = options.name;
+    exOpt.factory = factory;
+    if (this._modelFactories[name]) {
+      throw new Error(`Duplicate registered factory ${name}`);
+    }
+    this._modelFactories[name] = exOpt;
+    return new DisposableDelegate(() => {
+      delete this._modelFactories[name];
+    });
+  }
+
+  /**
+   * Get the list of registered widget factory display names.
+   *
+   * @param path - An optional file path to filter the results.
+   *
+   * #### Notes
+   * The first item in the list is considered the default.
+   */
+  listWidgetFactories(path?: string): string[] {
+    let ext = '.' + path.split('.').pop();
+    let factories: string[] = [];
+    let options: Private.IWidgetFactoryEx;
+    let name = '';
+    // If an extension was given, filter by extension.
+    // Make sure the modelFactory is registered.
+    if (ext.length > 1) {
+      if (ext in this._defaultWidgetFactories) {
+        name = this._defaultWidgetFactories[ext];
+        options = this._widgetFactories[name];
+        if (options.modelName in this._modelFactories) {
+          factories.push(name);
+        }
+      }
+    }
+    // Add the default widget if it was not already added.
```

Comment:
Do you see the global default as a fallback for all the openers, or as a fallback for the default openers? If it's a fallback for all openers, then it should be *last* in the list, not first. Otherwise your `.*` registration takes precedence over any other widgets that can handle the file.